pkg/kubelet/util/util_unix.go

```diff
@@ -27,6 +27,7 @@ import (
 	"time"
```

Comment:
Can you provide a little more background on your decision to place this back off in the generic `CreateListener` function, instead of `pkg/kubelet/server/server.go#ListenAndServePodResources`?

Specifically, I'm curious if we will always want this behavior when we create a listener, or if it is more of special case behavior because, for pod resources, a different process is performing important initialization steps?

If we do decide to keep this in `util_unix.go`, do we also want similar behavior in `util_windows.go`?

Reply:
> Can you provide a little more background on your decision to place this back off in the generic `CreateListener` function, instead of `pkg/kubelet/server/server.go#ListenAndServePodResources`?
> 

The `CreateListener` returns other errors too(protocol parsing error or unsupported protocols) and we should not put wait on those errors as well. Hence, we needed to put the wait as close to the actual creation time of the Listener object where the path error is triggered.
 
> Specifically, I'm curious if we will always want this behavior when we create a listener, or if it is more of special case behavior because, for pod resources, a different process is performing important initialization steps?

That's correct. The initialization of the pod-resources path is handled by some other thread. This wouldn't be necessary if we could ensure that the path for the socket is there already.

> 
> If we do decide to keep this in `util_unix.go`, do we also want similar behavior in `util_windows.go`?

Thanks for high-lighting this. I've added the changes for `util_unix.go` now as well.